fleetshard/pkg/central/reconciler/init_auth.go

```diff
@@ -157,9 +157,17 @@ func createAuthProviderRequest(central private.ManagedCentral) *storage.AuthProv
 	return request
 }
 
-func authProviderName(central private.ManagedCentral) string {
-	return ternary.String(strings.Contains(central.Spec.Auth.Issuer, "stage"),
-		"Red Hat SSO (Stage)", "Red Hat SSO")
+// authProviderName deduces auth provider name from issuer URL.
+func authProviderName(central private.ManagedCentral) (name string) {
+	switch {
+	case strings.Contains(central.Spec.Auth.Issuer, "sso.stage.redhat"): name = "Red Hat SSO (stage)"
+	case strings.Contains(central.Spec.Auth.Issuer, "sso.redhat"): name = "Red Hat SSO"
+	default: name = urlfmt.GetServerFromURL(central.Spec.Auth.Issuer)
```

Comment:
```suggestion
	case strings.Contains(central.Spec.Auth.Issuer, "sso.stage.redhat"):
		name = "Red Hat SSO (stage)"
	case strings.Contains(central.Spec.Auth.Issuer, "sso.redhat"):
		name = "Red Hat SSO"
	default:
		name = urlfmt.GetServerFromURL(central.Spec.Auth.Issuer)
```
imho much easier to read